src/ServicePulse.Host/app/js/views/message/controller.js

```diff
@@ -51,7 +52,7 @@
 
             if (!angular.isDefined(message.messageBody)) {
                 serviceControlService.getMessageBody(message.message_id).then(function (msg) {
-                    message.messageBody = msg.data;
+                    message.messageBody = prettifyService.prettifyText(msg.data);
```

Comment:
does the `msg` object maybe have the `content-type` property on it that we can use to check if the object is json or xml? Thereby bypassing the `try...catch` logic.

Reply:
I need to verify it, as that would be super useful.

Reply:
Ok so the situaion looks as follows: when we make a call to get body of a message it sets content type well to application/json or text/xml depending on the type. However when you ask again for the body then the 304 is returned and becasue of this line:
https://github.com/Particular/ServiceControl/blob/master/src/ServiceControl/Infrastructure/Nancy/NotModifiedStatusExtension.cs#L42

The contenty type is changed to `text/html` (default content type for Response object in Nancy. Even that before this line `context.Response.ContentType' has the correct value, but it is overwriten by the default html one. So to make this mechanism to work we would have to change that line of code to something as follows:

```
context.Response = new Response {StatusCode = HttpStatusCode.NotModified, ContentType = context.Response.ContentType};
```

Reply:
That's looking at the content-type of the http response, we need to look at the content type of the actual underlying message which would never change with nancy's pipeline. We want the `NServiceBus.ContentType` header from the actual message object